File: android/app/src/main/java/skglobal/product/flutter_uv_blind_refactor/methodchannels/DeeplinkMethodChannel.java
package skglobal.product.flutter_uv_blind_refactor.methodchannels;

import android.net.Uri;

import io.flutter.plugin.common.BinaryMessenger;
import io.flutter.plugin.common.MethodChannel;

public class DeeplinkMethodChannel {
    private MethodChannel channel;

    private DeeplinkMethodChannel() {
    }

    public static DeeplinkMethodChannel getInstance() {
        return SingletonHelper.INSTANCE;
    }

    public void register(BinaryMessenger messenger) {
        channel = new MethodChannel(messenger, "uv_deeplink");
        channel.setMethodCallHandler(handler);
    }

    private final MethodChannel.MethodCallHandler handler = (call, result) -> result.notImplemented();

    public void openLink(Uri uri) {
        if (uri != null) {
            channel.invokeMethod("open_link", uri.toString());
        }
    }

    private static class SingletonHelper {
        private static final DeeplinkMethodChannel INSTANCE = new DeeplinkMethodChannel();
    }
}

File: ios/Runner/methodchannels/DeeplinkMethodChannel.swift
//
//  DeeplinkMethodChannel.swift
//  Runner
//
//  Created by Tuna on 7/26/21.
//

import Foundation

class DeeplinkMethodChannel {
    static let shared = DeeplinkMethodChannel()
    private var channel: FlutterMethodChannel?
    
    private init() {}
    
    func register(with registrar: FlutterPluginRegistrar)  {
        channel = FlutterMethodChannel(name: "uv_deeplink", binaryMessenger: registrar.messenger())
        channel?.setMethodCallHandler(handler)
    }
    
    func handler(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        result(FlutterMethodNotImplemented)
    }
    
    func openLink(url: URL?) {
        guard let urlString: String = url?.absoluteString else {
            return
        }

        channel?.invokeMethod("open_link", arguments: urlString)
    }
}
